modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -1425,9 +1430,10 @@ void WindowContextTop::restack(bool restack) {
 
 void WindowContextTop::set_modal(bool modal, WindowContext* parent) {
     if (modal) {
-        //gtk_window_set_type_hint(GTK_WINDOW(gtk_widget), GDK_WINDOW_TYPE_HINT_DIALOG);
         if (parent) {
             gtk_window_set_transient_for(GTK_WINDOW(gtk_widget), parent->get_gtk_window());
+        } else {
+            gtk_window_set_keep_above(GTK_WINDOW(gtk_widget), modal ? TRUE : FALSE);
```

Comment:
The test of `modal` is unnecessary here, since this is all within an `if (modal)` block.

Reply:
Done.